### PR TITLE
allow filtering nodes by list of statuses

### PR DESCRIPTION
**Pull request by Omarabdul3ziz**
### Description
allow filtering nodes by list of statuses

### Changes
- change the status field type in node/farm/stats filters to be slice
- update the node status decision for sql condition
- update tests to cover the new changes
- update node status filtering at the other packages

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/1010

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

**Review by ashraffouda — APPROVED** [review]

File: grid-proxy/pkg/client/grid_client_test.go [hunk 1]
```diff
@@ -344,12 +343,11 @@ func testSuccess(t *testing.T, f ProxyFunc) {
 }
 
 func TestPrepareURL(t *testing.T) {
-	status := "st"
 	freeMRU := uint64(10)
 	farmIDs := []uint64{1, 2, 3}
 	dedicated := true
 	filter := types.NodeFilter{
-		Status:    &status,
+		Status:    []string{"st"},
```
Comment (xmonader) [inline grid-proxy/pkg/client/grid_client_test.go:350]: Can you please explain this test? I don't get what it does

File: grid-proxy/tests/queries/utils.go [hunk 2]
```diff
@@ -59,3 +59,12 @@ func SerializeFilter[F Filter](f F) string {
 
 	return res
 }
+
+func getRandomSlice[T any](slice []T) []T {
```
Comment (xmonader) [inline grid-proxy/tests/queries/utils.go]: Where is that used?

Comment (xmonader) [reply grid-proxy/tests/queries/utils.go]: I also would pass the number of elements that I need in the slice as a parameter with a guard clause. And maybe rename it to `RandomSliceFrom`

File: grid-proxy/tests/queries/utils.go [hunk 3]
```diff
@@ -59,3 +59,12 @@ func SerializeFilter[F Filter](f F) string {
...
+	randomLen := rand.Intn(len(slice))
```
Comment (xmonader) [inline grid-proxy/tests/queries/utils.go]: I don't see a rand.Seed call?

File: grid-proxy/pkg/client/grid_client_test.go (hunk 1 above)
Comment (Omarabdul3ziz) [reply grid-proxy/pkg/client/grid_client_test.go:350]: it validates the output of the `prepareUrl` function. by making sure the `got` url path/query - returned from the `prepareUrl` function - equals to the `want` url path/query - created from a literal string -

so having a filter
```go
types.NodeFilter{
    Status:    []string{"st"}
}
```
should output `?status=st` and so on

Comment (Omarabdul3ziz) [reply grid-proxy/pkg/client/grid_client_test.go:350]: it validates the output of the `prepareUrl` function. by making sure the `got` url path/query - returned from the `prepareUrl` function - equals to the `want` url path/query - created from a literal string -

so having a filter
```go
types.NodeFilter{
    Status:    []string{"st"}
}
```
should output `?status=st` and so on

File: grid-proxy/tests/queries/utils.go (hunk 3 above)
Comment (Omarabdul3ziz) [reply grid-proxy/tests/queries/utils.go]: found it here. it is based on a flag
https://github.com/threefoldtech/tfgrid-sdk-go/blob/3358c61e65e67e157b5ce497ceb9799b86c101da/grid-proxy/tests/queries/main_test.go#L60

**Pull request merged**